content/unexpected-reboots.md

```diff
@@ -0,0 +1,441 @@
+tableOfContents: true
+---
+
+If your system is spontaneously rebooting, this article provides steps to isolate the cause.
+
+> **NOTE:** There is a subtle but important difference between a reboot and a shutdown/power-off. In a reboot, parts of the system are left powered on to facilitate restarting the system. In a shutdown/power-off, hardware from all parts of the system except charging circuitry and capacitors is powered down.
+
+## Troubleshooting (Device Agnostic)
+
+### Software Causes
+
+Anything not directly affected by hardware, is controlled by software. This includes the kernel, drivers, and the desktop environment. The Operating System (OS), is essentially a software program that controls other software programs.
+
+If anything running in software causes a large enough cascade effect, it can take the rest of the system down and cause reboots or shut-offs.
+
+Software factors that can cause spontaneous reboots or crashes, include but are not limited to:
+
+1. Kernel Panics
+2. Kernel or Desktop Process Crashes
+3. Application Memory Leaks
+4. Driver Errors
+5. User Configuration Error
+6. Corrupted Operating System Files
+
+### Possible Solutions
+
+<!-- To be added ASAP
+
+**Upgrade or Reinstall the Kernel**
+
+[LINK]
+
+-->
+
+#### Check for Memory Leaks
+
+If an app is taking more than its fair share of system memory, that can have a runaway affect resulting in shutdowns or restarts.
+
+Applications are available to help diagnose memory leaks.
+
+From the `Terminal` run the `top` command to see which processes are using the most system resources:
+
+Open a `Terminal` with <kbd>SUPER</kbd>+<kbd>T</kbd> (Pop!\_OS) or <kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>T</kbd> (Ubuntu) run the following commands:
+
+```bash
+top
+```
+
+If a GUI tool is preferred, and you are running PopOS or Ubuntu, open "System Monitor":
+
+![system-monitor](/images/unexpected-reboots/system-monitor.png)
+
+#### Remove or Reinstall Drivers
+
+If a package is misbehaving or corrupted, try reinstalling it, or removing it. This can be done with the following `Terminal` commands.
+
+> **NOTE:** replace [packagename] with the same of the application or driver you are are trying to reinstall. Do not include the "[ ]" around the name.
+
+```bash
+sudo apt update
+sudo apt install --reinstall [packagename]
+```
+
+To more completely remove the package:
+
+```bash
+sudo apt update
+sudo apt remove [packagename]
+sudo apt clean
+sudo apt autoremove
+```
+
+for more info, check out our [Package Management Pop!_OS Article](/articles/package-manager-pop) or the [Ubuntu Equivalent](/articles/package-manager-ubuntu) 
+
+#### Test Admin User
+
+Sometimes an intentional or unintentional change to system configuration files can cause unexpected boot behavior, and is only limited to the user that made the changes.
+
+To test whether the problem exists at a user-level or system level, create a test admin user and see if the reboots or shutdowns occur there.
```

Comment:
It looks like "user-level" should not be hyphenated ("system level" is not hyphenated.)

Reply:
I think in this case it makes more sense to hyphenate both, since we say "user-level" issues, or "system-level" issues, but I don't know if it's considered grammatically correct.

Reply:
Fixed.